datacube/utils/dask.py

```diff
@@ -12,6 +12,7 @@
 import queue
 from dask.distributed import Client
 import dask
+from dask.delayed import delayed
```

Comment:
The [dask delayed API reference uses](https://docs.dask.org/en/stable/delayed-api.html) `from dask import delayed`.

I'm sure this is fine too. :)